src/utils.js

```diff
@@ -236,6 +236,84 @@ function camelcase (dashCaseText) {
     })
 }
 
+// We use this method to detect M1 macs and set appropriate API values to prevent sites from detecting fingerprinting protections
+function isAppleSilicon () {
+    const canvas = document.createElement('canvas')
+    const gl = canvas.getContext('webgl')
+
+    // Best guess if the device is an Apple Silicon
+    // https://stackoverflow.com/a/65412357
+    return gl.getSupportedExtensions().indexOf('WEBGL_compressed_texture_etc') !== -1
+}
+
+/**
+ * Take configSeting which should be an array of possible values.
+ * If a value contains a criteria that is a match for this environment then return that value.
+ * Otherwise return the first value that doesn't have a criteria.
+ *
+ * @param {array} configSetting - Config setting which should contain a list of possible values
+ * @returns - The value from the list that best matches the criteria in the config
```

Comment:
```suggestion
 * @param {*[]} configSetting list of possible config options
 * @returns {*|undefined} The value from the list that best matches the criteria in the config
```

Optionally you could setup a type for this something like:
```
@typedef ConfigOption
@property {Record<string, *>} critera
@property {string} type
@property {*} value
```

And then instead:
```suggestion
 * @param {ConfigOption[]} configSetting list of possible config options
 * @returns {ConfigOption|undefined} The value from the list that best matches the criteria in the config
```

(See other comments, I think this function should return _undefined_ when the match isn't found as an error state)

Another note is that tsconfig.json should include this file as we're adding some typings (I checked locally and nothing goes wrong by adding it so it's worth including so that automation catches issues when someone edits the file)
 - FYI: We're slowly rolling out typing, so it's an opt in for extension and content-scope-scripts currently.